conformance/tests/directives_deprecated.py

```diff
@@ -0,0 +1,108 @@
+"""
+Tests the warnings.deprecated function.
+"""
+
+# pyright: reportDeprecated=true
+
+# Specification: https://typing.readthedocs.io/en/latest/spec/directives.html#deprecated
+# See also https://peps.python.org/pep-0702/
+
+# > Type checkers should produce a diagnostic whenever they encounter a usage of an object
```

Comment:
There's an open question in the spec about whether decorators applied on top of a `@deprecated` decorator should retain the deprecation message — and under what circumstance. 

```python
def other_decorator[**P, R](c: Callable[P, R]) -> Callable[P, R]: ...

@other_decorator
@deprecated("???")
def func() -> None: ...
```

In pyright's implementation, I chose to record the deprecated message as part of the function's signature, and this signature is then bound to a ParamSpec. This retains the deprecated message in cases where the deprecated function is passed through other decorators. The spec isn't really clear about this behavior, so maybe it's best to leave out of the conformance suite for now. Maybe @JelleZijlstra has thoughts about this.

Reply:
I feel the behavior users would expect is that the deprecation gets preserved.